### PR TITLE
doc(data): add example ngrx data app to repo

### LINKED ISSUE #2868: ngrx/data example app
Please, add a reference example using ngrx/data, just like https://github.com/ngrx/platform/tree/master/projects/example-app, but using all ngrx/data functions.
Having an example of reference use, you can understand the general principle and best practices, and refer to the documentation only for specific clarifications. The documentation, as it is, does not provide a general understanding of use for beginners. It provides background information that is needed by those who already know how to use your library.

### PR DESCRIPTION
**Pull request by yharaskrik**
Closes: #2868

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ x Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

Closes #2868

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/data-example-app/src/board/board.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<ngrx-data-nestjs-minimal-boilerplate-kanban-board-ui
```
Comment (brandonroberts) [inline projects/data-example-app/src/board/board.component.html:1]:
```suggestion
<ngrx-data-kanban-board-ui
```

File: projects/data-example-app/src/board/board.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@Component({
+  selector: 'board-component',
+  templateUrl: './board.component.html',
+  styleUrls: ['./board.component.scss'],
+})
+export class BoardComponent {
+  stories$: Observable<Stories[]> = this.storyDataService.groupedStories$;
+
+  constructor(private storyDataService: StoryDataService) {
+    this.storyDataService.getAll();
```
Comment (brandonroberts) [inline projects/data-example-app/src/board/board.component.ts:15]: Move to ngOnInit

**Review by brandonroberts — CHANGES_REQUESTED** [review]
nits

File: projects/data-example-app/src/board/board.component.html (hunk 1 above)
Comment (yharaskrik) [reply projects/data-example-app/src/board/board.component.html:1]: Whoops forgot to change the selector!

File: projects/data-example-app/src/board/board.component.ts (hunk 2 above)
Comment (yharaskrik) [reply projects/data-example-app/src/board/board.component.ts:15]: Will do!

**Comment by yharaskrik** [comment]
Need to wait for https://github.com/ngrx/platform/pull/2906 as I butchered the typings for the NgRx Data add method.

**Comment by yharaskrik** [comment]
This should be good to go now. It is basic but we can add more to it to show off how to use the different features of ngrx data.

**Comment by timdeschryver** [comment]
I'll also give this a review later this week, thanks in advance @yharaskrik  🙌.

File: projects/data-example-app/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      ],
+    },
+  },
+  coverageDirectory: '../../coverage/projects/data-example-app',
+  snapshotSerializers: [
+    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
+    'jest-preset-angular/build/AngularSnapshotSerializer.js',
+    'jest-preset-angular/build/HTMLCommentSerializer.js',
+  ],
+};
```
Comment (timdeschryver) [inline projects/data-example-app/jest.config.js:21]: To keep this inline with the other configs:
- add a displayName
- move coverage to apps
- use the "new" syntax to set `astTransformers`

```suggestion
module.exports = {
  name: 'data-example-app',
  displayName: 'Data Example App',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/apps/data-example-app',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
  setupFilesAfterEnv: ['<rootDir>/src/setup-jest.ts'],
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
      stringifyContentPathRegex: '\\.(html|svg)$',
      astTransformers: {
        before: [
          'jest-preset-angular/build/InlineFilesTransformer',
          'jest-preset-angular/build/StripStylesTransformer',
        ],
      },
    },
  },
};
```

File: projects/data-example-app/src/app/app.component.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'ngrx-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
+})
+export class AppComponent {
+  title = 'data-example-app';
```
Comment (timdeschryver) [inline projects/data-example-app/src/app/app.component.ts:9]:
```suggestion
```

File: projects/data-example-app/src/board/ui/board-ui.component.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  @Input() stories: Stories[] = [];
+
+  @Output() add = new EventEmitter<Story>();
+
+  @Output() delete = new EventEmitter<DeleteStoryDto>();
+
+  @Output() update = new EventEmitter<UpdateStoryDto>();
+
+  addNew(column: number, stories: Stories): void {
+    // `as any` because of: https://github.com/ngrx/platform/pull/2899
```
Comment (timdeschryver) [inline projects/data-example-app/src/board/ui/board-ui.component.ts:25]: Is this stil the case?

**Review by timdeschryver — COMMENTED** [review]
LGTM, I left a few small comments.
I think we should also add a README with a guide how to serve this example app.

File: projects/data-example-app/src/board/ui/board-ui.component.ts (hunk 5 above)
Comment (yharaskrik) [reply projects/data-example-app/src/board/ui/board-ui.component.ts:25]: Yes it is to my knowledge, The pr was merged but I don't know if it has been released yet?

Comment (timdeschryver) [reply projects/data-example-app/src/board/ui/board-ui.component.ts:25]: Yes, these changes are included in the v11 releases 
Side note: the examples are not using the published packages, but are using the sources directly - so we're always on the latest version here 🙂

Comment (yharaskrik) [reply projects/data-example-app/src/board/ui/board-ui.component.ts:25]: great sounds good! When I wrote this my PR hadn't been merged in yet but I can update this code now!

**Comment by yharaskrik** [comment]
hey @timdeschryver sorry it took so long but I made those changes! Should be good to go, wasn't sure what all to include in the readme.

**Comment by timdeschryver** [comment]
Thanks @yharaskrik, this is looking good to me!

 I notice that the tests are failing (probably because we recently changed the setup), I'll take a look at it tomorrow so we can merge this in soon. Thanks again 🎉

**Comment by yharaskrik** [comment]
Ya it is the demo app that is failing, i can take a look at it today as well.

**Comment by yharaskrik** [comment]
@timdeschryver should be good to go! The setup-jest.ts file was missing apparently.

File: projects/data-example-app/jest.config.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  displayName: 'Data Example App',
+  preset: '../../jest.preset.js',
+  coverageDirectory: '../../coverage/apps/example-app',
```
Comment (timdeschryver) [inline projects/data-example-app/jest.config.js:4]:
```suggestion
  coverageDirectory: '../../coverage/apps/data-example-app',
```

**Comment by yharaskrik** [comment]
Awesome thanks! It is definitely the morning i did not notice the coverage directory change when i updated the jest config haha. Thanks!

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**